homeassistant/components/zwave/__init__.py

```diff
@@ -311,30 +321,20 @@ def value_added(node, value):
 
     def node_added(node):
         """Handle a new node on the network."""
-        entity = ZWaveNodeEntity(node, network)
-        node_config = device_config.get(entity.entity_id)
+        entity = ZWaveNodeEntity(node, network, new_entity_ids)
+        name = node_name(node)
+        if new_entity_ids:
+            generated_id = generate_entity_id(DOMAIN + '.{}', name, [])
+        else:
+            generated_id = entity.entity_id
+        node_config = device_config.get(generated_id)
```

Comment:
Since the ID list in line 327 is empty it will just slugify it without collision prevention.
So in case of collision both device will have the same ID here pulling the same config.

Also, since rename_value acts by value_id which is not exposed to the UI in any way, does user using the UI rename card has a way to determine which of the two identical entities they are renaming?